controller/Controller.py

```diff
@@ -1,25 +1,68 @@
-import time
+import sys
 import logging
+import pygame
+import view.ViewConstants as props
+from view.MainMenu import MainMenu
+from controller.ControllerView import ControllerGameView, ControllerMainMenu
 
 
-
-class Controller:
+class Controller(ControllerGameView, ControllerMainMenu):
+    """
+    class implementing a basic controller
+    """
 
     def __init__(self):
-        pass
+        # call init of ControllerGameView
+        super(Controller, self).__init__()
```

Comment:
@csacro  kannst du mir erklären, wie du das mit den Super Calls meintest. Verstehe das noch nicht so ganz..?

Reply:
Du erbst ja von zwei Klassen. Wenn du nur `super().__init__()` machst, entspricht das nur `super(Controller, self).__init__()` und damit wird nur die init von ControllerGameView, also der ersten Super-Klasse aufgerufen.
Durch `super(ControllerGameView, self).__init__()` wird die init von ControllerMainMenu aufgerufen, also der Super-Klasse nach ControllerGameView.
Dies liegt an MRO (Class search path).

Reply:
mMn. sollte das so sein wie in 24cb5c7

Reply:
Man sollte doch beide Super-Konstruktoren aufrufen, wenn man von zwei Klassen erbt, oder nicht?

Reply:
https://stackoverflow.com/questions/3277367/how-does-pythons-super-work-with-multiple-inheritance

Reply:
Ok ich versteh das falsch nvm